tests/Simplify/EqualTest.elm

```diff
@@ -934,6 +934,787 @@ a = ({ a = 1 }).a == ({ a = 1 }).b
+            \() ->
+                """module A exposing (..)
+a : Bool
+a = 0 /= (l |> List.length)
+"""
+                    |> Review.Test.run ruleWithDefaults
+                    |> Review.Test.expectErrors
+                        [ Review.Test.error
+                            { message = "This can be replaced with a call to (not << List.isEmpty)"
+                            , details =
+                                [ "Whereas List.length takes as long to run as the number of elements in the List,"
+                                , "List.isEmpty runs in constant time."
+                                ]
+                            , under = "0 /= (l |> List.length)"
+                            }
+                            |> Review.Test.whenFixed """module A exposing (..)
+a : Bool
+a = (l |> (not << List.isEmpty))
+"""
+                        ]
+        ]
+
+
+setTests : Test
+setTests =
+    describe "Set.size should be Set.isEmpty"
+        [ test "should replace Set.size s == 0 with Set.isEmpty s" <|
+            \() ->
+                """module A exposing (..)
+a : Bool
+a = Set.size s == 0
+"""
+                    |> Review.Test.run ruleWithDefaults
+                    |> Review.Test.expectErrors
+                        [ Review.Test.error
+                            { message = "This can be replaced with a call to Set.isEmpty"
+                            , details =
+                                [ "Whereas Set.size takes as long to run as the number of elements in the Set,"
+                                , "Set.isEmpty runs in constant time."
+                                ]
+                            , under = "Set.size s == 0"
+                            }
+                            |> Review.Test.whenFixed """module A exposing (..)
+a : Bool
+a = Set.isEmpty s
+"""
+                        ]
+        , test "should replace size s == 0 with isEmpty s" <|
+            \() ->
+                """module A exposing (..)
+import Set as CoreSet exposing (..)
+
+
+a : Bool
+a = size s == 0
+"""
+                    |> Review.Test.run ruleWithDefaults
+                    |> Review.Test.expectErrors
+                        [ Review.Test.error
+                            { message = "This can be replaced with a call to isEmpty"
+                            , details =
+                                [ "Whereas size takes as long to run as the number of elements in the Set,"
+                                , "isEmpty runs in constant time."
+                                ]
+                            , under = "size s == 0"
+                            }
+                            |> Review.Test.whenFixed """module A exposing (..)
+import Set as CoreSet exposing (..)
+
+
+a : Bool
+a = isEmpty s
+"""
+                        ]
+        , test "should replace 0 == Set.size s with Set.isEmpty s" <|
+            \() ->
+                """module A exposing (..)
+a : Bool
```

Comment:
There's technically a missing import here (and the same in other tests).

Reply:
good call. Will update.